rust/agama-dbus-server/src/network/system.rs

```diff
@@ -73,6 +73,16 @@ impl<T: Adapter> NetworkSystem<T> {
             Action::UpdateConnection(conn) => {
                 self.state.update_connection(conn)?;
             }
+            Action::UpdateControllerConnection(conn, settings, tx) => {
+                let id = &conn.clone();
```

Comment:
Why do you need to clone the conn? I am not saying that it is wrong, but looks weird.